garden-backend-service/src/sandboxed_functions/modal_publishing_helpers.py

```diff
@@ -41,13 +43,22 @@ def get_app_from_file_contents(file_contents: str):
 
 def validate_modal_file(file_contents: str):
     user_app = get_app_from_file_contents(file_contents)
-    function_names = [f for f in user_app.registered_functions if "*" not in f]
     app_name = user_app.name
-    return {"function_names": function_names, "app_name": app_name}
+    functions = get_function_specs(user_app.registered_functions)
+    return {"app_name": app_name, "functions": functions}
 
     # TODO: confirm nothing dastardly on the app/functions
 
 
+def get_function_specs(functions: dict[str, modal.Function]):
+    funcs = {name: dataclasses.asdict(func.spec) for name, func in functions.items()}
+    for _, spec in funcs.items():
+        # these fields aren't easily serializable, I don't think we need them
+        del spec["image"]
+        del spec["mounts"]
```

Comment:
I don't know about sending over the whole function spec less these fields. I think we should instead identify the exact fields we expect/need and pull them out. And maybe throw an error in here that will let us know if/when Modal changes the internal format of the hardware spec